### PR TITLE
CTDA-1669: Add metrics in the PPNS connector, fraction of posted messages that have boxes attached

### PR DESCRIPTION
**Pull request by drnaylor**
Thoughts:

* ~~I suspect what we're interested in is the ratio of calls to getBox vs sendPushNotification over the entire lifetime of the application. In this case, a ratio of two counters is what we want. However, if we want it over a rolling time period, we might be able to do this with Meters.~~
* ~~Do we just want a ratio of calls to these methods, or do we want a ratio of _successful_ calls? My interpretation is the former, but if it's the latter, should be easy to do.~~

Departures will follow once we're happy with this.

---

This has been through a lot of iterations, we settled on the fraction of submitted arrivals that have a box attached. We also have a week sliding window for this data.

### CODE REVIEW

**Review by RobertLaverick — APPROVED** [review]

**Comment by drnaylor-oc** [comment]
The two commits in this PR at this stage should be considered as two separate solutions.

**Review by RobertLaverick — APPROVED** [review]

**Comment by RobertLaverick** [comment]
test me again

**Comment by RobertLaverick** [comment]
retest this please

**Comment by drnaylor-oc** [comment]
retest

File: app/audit/AuditDetails.scala [hunk 1]
```diff
@@ -18,20 +18,30 @@ package audit
...
+}
+
 sealed abstract class AuditDetails {
   def channel: ChannelType
   def customerId: String
   def json: JsObject
 }
 
-case class AuthenticatedAuditDetails(channel: ChannelType, enrolmentId: Ior[TURN, EORINumber], json: JsObject) extends AuditDetails {
+case class AuthenticatedAuditDetails(channel: ChannelType, enrolmentId: Ior[TURN, EORINumber], json: JsObject, boxId: Option[BoxId]) extends AuditDetails {
```
Comment (RobertLaverick) [inline app/audit/AuditDetails.scala:44]: have you considered having boxId default to None?
`boxId: Option[BoxId] = None`

**Review by RobertLaverick — CHANGES_REQUESTED** [review]

**Review by RobertLaverick — APPROVED** [review]

**Pull request merged**